sippy-ng/src/component_readiness/CompReadyTestReport.js

```diff
@@ -251,30 +231,42 @@ export default function CompReadyTestReport(props) {
       return itemB - itemA
     })
 
+    if (!versions[version]) {
+      // Handle the case where GA date is undefined (implies under development and not GA)
+      const today = new Date()
+      const weeksBefore = Math.floor(
+        (today - fromDate) / (1000 * 60 * 60 * 24 * 7)
+      )
+      return `Recent ${weeksBefore} week(s) of ${version}`
+    }
```

Comment:
Not quite :-) In my previous example it would report something like "Recent 121 weeks of 4.13" now


Maybe something like this?

```javascript
// Handle the case where GA date is undefined (implies under development and not GA)
const weeksBefore = Math.floor((toDate - fromDate) / (1000 * 60 * 60 * 24 * 7));

// Calculate the difference between now and toDate in hours
const now = new Date();
const hoursDifference = Math.abs(now - toDate) / (1000 * 60 * 60);

if (hoursDifference <= 72) {
    return `Recent ${weeksBefore} week(s) of ${version}`
} else {
    const toDateFormatted = new Date(toDate).toLocaleDateString(); // Convert toDate to human-readable format
    return `${weeksBefore} week(s) before ${toDateFormatted} of ${version}`
}
```

Reply:
Thanks for that! added.